pkg/storage/tpr/storage_interface_test.go

```diff
@@ -71,20 +74,245 @@ func TestCreate(t *testing.T) {
 	if obj == nil {
 		t.Fatal("no broker was in storage")
 	}
-	if !reflect.DeepEqual(outBroker, obj) {
+	if !reflect.DeepEqual(createdBroker, obj) {
 		t.Fatalf(
 			"output and object in storage are different: %s",
-			diff.ObjectReflectDiff(outBroker, obj),
+			diff.ObjectReflectDiff(createdBroker, obj),
 		)
 	}
 	// Output and what's in storage should be known to be deeply equal at this
 	// point. Compare either of those to what was passed in. The only diff should
-	// be resource version, so we will clear that first.
-	outBroker.ResourceVersion = ""
-	if !reflect.DeepEqual(broker, outBroker) {
+	// be resource version, so we will set that first.
+	origBroker.ResourceVersion = createdBroker.ResourceVersion
+	if !reflect.DeepEqual(origBroker, createdBroker) {
 		t.Fatalf(
 			"input and output are different: %s",
-			diff.ObjectReflectDiff(broker, outBroker),
+			diff.ObjectReflectDiff(origBroker, createdBroker),
+		)
+	}
+	// End create test
+
+	// Begin create existing test
+	origBroker2 := &sc.Broker{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+	}
+	createdBroker2 := &sc.Broker{}
+	err = iface.Create(
+		context.Background(),
+		key,
+		origBroker2,
+		createdBroker2,
+		uint64(0),
+	)
+	if err == nil {
+		t.Fatal("expected error when creating existing resource, but got none")
+	}
+	storageErr, ok := err.(*storage.StorageError)
```

Comment:
These next ~25 lines are similar to later ones in this file, could we hoist them into a method?
verifyError(err, storage.StorageError, storage.ErrCodeKeyExists)

or something like that, where the verify method would then have these lines there since they all seem very similar?